packages/stack/contracts-manager/test/contracts-manager.spec.js

```diff
@@ -0,0 +1,199 @@
+import ContractsManager from '../src';
+import Contract from '../src/contract';
+
+import sinon from 'sinon';
+import assert from 'assert';
+import { fakeEmbark } from 'embark-testing';
+
+const { embark, plugins } = fakeEmbark();
+
+// Due to our `DAPP_PATH` dependency in `embark-utils` `dappPath()`, we need to
+// ensure that this environment variable is defined.
+process.env.DAPP_PATH = 'something';
+
+describe('stack/contracts-manager', () => {
+
+  let contractsManager;
+
+  beforeEach(() => {
+    contractsManager = new ContractsManager(embark, { plugins });
+  });
+
+  afterEach(() => {
+    embark.teardown();
+    sinon.restore();
+  });
+
+  it('should add contract objects', async () => {
+    const testContract = {
+      className: 'TestContract'
+    };
+
+    await embark.events.request2('contracts:add', testContract);
+    const contract = await embark.events.request2('contracts:contract', testContract.className);
+    assert.equal(contract.className, testContract.className);
+  });
+
+  it('should return list of added contract objects', async () => {
+    const testContract = {
+      className: 'TestContract'
+    };
+
+    await embark.events.request2('contracts:add', testContract);
+    const list = await embark.events.request2('contracts:list');
+    assert.equal(list.length, 1);
+    assert.equal(list[0].className, testContract.className);
+  });
+
+  it('should build contracts', async (done) => {
+
+    const contractsConfig = {
+      contracts: {
+        deploy: {
+          TestContract: {}
+        }
+      }
+    };
+
+    const compiledContracts = {
+      TestContract: {
+        // `compiledContract` can't be an empty object otherwise
+        // contracts-manager will exclude it from processing, resulting
+        // in `TestContract` not being managed.
+        code: 'some code',
+      }
+    };
+
+    const beforeBuildAction = sinon.spy((params, cb) => { cb(null, params); });
```

Comment:
Sinon fakes have a [`yields` function](https://sinonjs.org/releases/v8.1.1/fakes/) that does exactly this:
```suggestion
    const beforeBuildAction = sinon.fake.yields(null, params);
```

Reply:
That's cool! Didn't know about that.

I believe though it won't do the trick here, because `params` isn't static.